packages/data/src/redux-store/index.js

```diff
@@ -252,10 +259,24 @@ export default function createReduxStore( key, options ) {
 				);
 			}
 
-			function bindMetadataSelector( selector ) {
+			function bindMetadataSelector( metaDataSelector ) {
 				const boundSelector = ( ...args ) => {
 					const state = store.__unstableOriginalGetState();
-					return selector( state.metadata, ...args );
+
+					const originalSelectorName = args && args[ 0 ];
+					const originalSelectorArgs = args && args[ 1 ];
+					const targetSelector =
+						options?.selectors?.[ originalSelectorName ];
+
+					// Normalize the arguments passed to the target selector.
+					if ( originalSelectorName && targetSelector ) {
+						args[ 1 ] = normalize(
+							targetSelector,
+							originalSelectorArgs
+						);
+					}
+
+					return metaDataSelector( state.metadata, ...args );
```

Comment:
One idea: the metadata selectors and the `metadata` state are private, not exposed in any public API, so we can change their parameters. As state, we could pass an object like:
```js
{
  metadata, // the actual state object
  normalizeArgs( selectorName, args ), // a new helper function
}
```
The `bindMetadataSelector` function has access to all registered selectors, their names and `normalizeArgs` functions. So, it can implement the `normalizeArgs` helper as:
```js
function normalizeArgs( selectorName, args ) {
  if ( allSelectors[ selectorName ]?.normalizeArgs ) {
    return allSelectors[ selectorName ].normalizeArgs( args );
  }
  return args;
}
```
and `getResolutionState` can be implemented as:
```js
function getResolutionState( state, selectorName, args ) {
  return state.metadata[ selectorName ].get( state.normalizeArgs( selectorName, args ) );
}
```

Reply:
This might be a "simpler" approach. I'll take a look.

Reply:
Looking here against. The complexity still exists in the `bindMetadataSelector` but we're just making the `getResolutionState` aware of the complexity as well.

I'm probably missing something but I'd like to understand more before I proceed with further refactor.